### ISSUE TITLE
TransactionTestCase wipes fixtures installed by data migrations

### ISSUE THREAD

**Issue by rhliang** [cfe-lab/Kive#500]
Running into this problem while working on #496.  The new-style execution code requires you to use a TransactionTestCase to run tests, as it uses multithreaded execution where each thread needs to access the database.

However, the procedure between tests appears to be a flush followed by a reload of the test fixtures.  This wipes the initial fixtures that are installed by our data migrations.  As such, the first test of the test case works, and the subsequent ones fail due to database-related errors.

We need a workaround for this behaviour.

**Comment by rhliang** [comment]
Evidently this is a known issue: https://code.djangoproject.com/ticket/23727

Looks like it's been fixed, so I'm going to act as our guinea pig and upgrade to Django 1.9 to see what happens.

**Comment by rhliang** [comment]
I upgraded to Django 1.9 (in a virtualenv), and also found that you're supposed to set `serialized_rollbacks=True` in the TestCase to help with this behaviour, but it just caused another, similar, problem:

```
IntegrityError: insert or update on table "metadata_basicconstraint" violates foreign key constraint "metadata_basiccons_datatype_id_29d239b1_fk_metadata_datatype_id"
DETAIL:  Key (datatype_id)=(5) is not present in table "metadata_datatype".
```

When Django serializes rollbacks, it deserializes the state of the database at the start of the TestCase into a string, and then deserializes the data from that string.  It appears the problem is that the contents of the serialized data is in the wrong order: all of the Datatypes are defined after the BasicConstraints, but the necessary Datatype (i.e. the one with primary key 5) is not defined by the time the BasicConstraint using it is.

**Comment by rhliang** [comment]
I tried to work around one of the cases where this is happening by breaking all of the individual tests in one of the offending TestCases into their own TestCases, but it didn't work either.  Whatever that first TransactionTestCase is doing to the database is futzing it up for the stuff that comes after.

**Comment by rhliang** [comment]
I partially addressed this with a workaround (see #513).  Now I'm able to run an entire TestCase, and even a whole suite at a time, but when I run the entire project's tests all at once, I get some that fail.  This is a bit reminiscent of the longstanding issue we had with leaking file descriptors, but I don't think that's the issue.

**Comment by rhliang** [comment]
The problem appeared to be that Django 1.9's test runner attempted to run the `kive/test_settings.py` file, which we used to run tests using SQLite.  (It probably looks for all files starting with `test`.)  I changed the names of `test_settings.py` and `test_settings_pg.py` to `settings_test_SQLite.py` and `settings_test_pg.py`, and that allows me to run the entire project's tests at once.
